app/controllers/organizations/tasks_controller.rb

```diff
@@ -1,28 +1,46 @@
 class Organizations::TasksController < Organizations::BaseController
-  skip_verify_authorized only: %i[cancel]
-
-  before_action :set_pet, only: %i[new create edit update destroy cancel]
-  before_action :set_task, only: %i[edit update destroy]
+  before_action :set_pet, only: %i[new show index create edit update destroy]
+  before_action :set_task, only: %i[show edit update destroy]
 
   def new
     authorize! Task, context: {pet: @pet}
 
     @task = @pet.tasks.build
   end
 
+  def index
+    authorize! Task, context: {pet: @pet}
+
+    if params[:cancel] == "true"
+      respond_to do |format|
+        format.html
+        format.turbo_stream { render turbo_stream: turbo_stream.update("new_task", "") }
+      end
+    end
+
+    @tasks = @pet.tasks.list_ordered
+  end
+
+  def show
+    respond_to do |format|
+      format.html
+      format.turbo_stream
+    end
+  end
+
   def create
     authorize! Task, context: {pet: @pet}
 
     @task = @pet.tasks.build(task_params)
 
     if @task.save
       respond_to do |format|
-        format.turbo_stream { render turbo_stream: turbo_stream.replace("tasks_list", partial: "organizations/pets/tasks/tasks", locals: {task: @task}) }
+        format.turbo_stream { render turbo_stream: turbo_stream.replace("tasks_pet_#{@pet.id}", partial: "organizations/pets/tabs/tasks", locals: {task: @task}) }
```

Comment:
Similar to `update`, I think we should be targeting the index, not the tab.

However, this made me realize I misguided you in my initial recommendation to move the stuff from the tabs to the task/index. Moving all the tasks stuff to the tasks views was the right move, but we should not have moved the entirety of `tabs/_tasks.html.erb` to `tasks/index.html/erb`. A lot of that html is specific to the tabs, and does not make sense for the tasks/index standing alone. And really, the turbo_frame for the pet tasks should encapsulate the rendered tasks only, not the whole template.

So instead, I think the better solution is this:
```erb
# move content from tasks/index.html.erb back to tabs/tasks
# and move the outermost turbo tags to only surround the tasks render

# app/views/organizations/pets/tabs/_tasks.html.erb
<div class="row">
  <div class="col-12 mb-4">
  ...
        <%= turbo_frame_tag dom_id(@pet, :tasks) do %>
          <%= render @tasks %>
        <% end %>
        ...
</div>

# app/views/organizations/tasks/index.html.erb
<%= turbo_frame_tag dom_id(@pet, :tasks) do %>
  <%= render @tasks %>
<% end %>

```
This will require `@tasks` to be defined in `PetsController#show`, which will also assist us in eager loading the tasks for the tab.
```ruby
# app/controllers/organizations/pets_controller.rb
  def show
    @active_tab = determine_active_tab
    @tasks = @pet.tasks
  end
```
And finally:
```erb
# app/views/organizations/tasks/create.turbo_stream.erb

<%= turbo_stream.update Task.new, "" %> 
<%= turbo_stream.append dom_id(@pet, :tasks), @task %> 
```
You can test things are working properly by opening an edit for one of the tasks, then open the new task button, and submit a new task. It should be appended at the end of the task list, and the edit form should remain open as well.

Let me know if you have any issues with this solution.

Reply:
Ignore this. See my reply to your comment on `update` https://github.com/rubyforgood/pet-rescue/pull/563/files#r1533089025

Again, sorry for getting ambitious in this review and missing the sorting issue!